src/main/java/jcifs/smb1/http/NetworkExplorer.java

```diff
@@ -0,0 +1,538 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs.smb1.http;
+
+import java.io.*;
+import javax.servlet.*;
+import javax.servlet.http.*;
+
+import jcifs.smb1.*;
+import jcifs.smb1.http.*;
+import jcifs.smb1.netbios.NbtAddress;
+import jcifs.smb1.smb1.*;
+import jcifs.smb1.util.Base64;
+import jcifs.smb1.util.LogStream;
+import jcifs.smb1.util.MimeMap;
+
+import java.util.*;
+import java.text.SimpleDateFormat;
+import java.net.UnknownHostException;
+
+/**
+ * This servlet may be used to "browse" the entire hierarchy of resources
+ * on an SMB network like one might with Network Neighborhood or Windows
+ * Explorer. The users credentials with be negotiated using NTLM SSP if
+ * the client is Microsoft Internet Explorer.
+ */
+
+public class NetworkExplorer extends HttpServlet {
+
+    private static LogStream log = LogStream.getInstance();
+
+    private MimeMap mimeMap;
+    private String style;
+    private NtlmSsp ntlmSsp;
+    private boolean credentialsSupplied;
+    private boolean enableBasic;
+    private boolean insecureBasic;
+    private String realm, defaultDomain;
+
+    public void init() throws ServletException {
+        InputStream is;
+        StringBuffer sb = new StringBuffer();
+        byte[] buf = new byte[1024];
+        int n, level;
+        String name;
+
+        Config.setProperty( "jcifs.smb1.smb.client.soTimeout", "600000" );
+        Config.setProperty( "jcifs.smb1.smb.client.attrExpirationPeriod", "300000" );
+
+        Enumeration e = getInitParameterNames();
+        while( e.hasMoreElements() ) {
+            name = (String)e.nextElement();
+            if( name.startsWith( "jcifs.smb1." )) {
+                Config.setProperty( name, getInitParameter( name ));
+            }
+        }
+
+        if( Config.getProperty( "jcifs.smb1.smb.client.username" ) == null ) {
+            ntlmSsp = new NtlmSsp();
+        } else {
+            credentialsSupplied = true;
+        }
+
+        try {
+            mimeMap = new MimeMap();
+            is = getClass().getClassLoader().getResourceAsStream( "jcifs/smb1/http/ne.css" );
+            while(( n = is.read( buf )) != -1 ) {
+                sb.append( new String( buf, 0, n, "ISO8859_1" ));
+            }
+            style = sb.toString();
+        } catch( IOException ioe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)